iOS-sample-app/Showcases/Screen Views/Modals/ProgrammaticModalPresentingViewController.swift

```diff
@@ -0,0 +1,49 @@
+
+import UIKit
+import ContentSquare
+
+// This class presents a modal programmatically. The modal is implemented in ProgrammaticModalViewController.
+//
+// We need to implement three screen views here:
+// - A screen view for this modal presenter when it appears on screen
+// - A screen view for the modal when it is presented on screen
+// - A screen view for this modal presenter when the modal is dismissed and this presenter is displayed again
+//
+// The first two are easy: we just send a screen view in the viewDidAppear of both the presenter and the modal. The
+// third is the real challenge here. When the modal is dismissed, we need to find a way to notify the presenter so that
+// it can send a screen view. We achieved this with a delegation pattern which is implemented with Dismissable and
+// DismissalDelegate. The mechanism can be reused with any pair of modal / modal presenter.
```

Comment:
Remember to verify this if you change the delegate structure to make sure it is still up to date.